### PR TITLE
DatabaseAccesser

### PR DESCRIPTION
**Pull request by turtle11235**
Database access is managed through a middleman access class

### CODE REVIEW

File: server/database/User.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+})
+
+var UserModel = mongoose.model('User', UserSchema );
+
+
+class User {
+
+    constructor(userIn){
+        this.validateUser(userIn)
+        if (userIn.isTeacher ){
```
Comment (ZOrangeBandit) [inline server/database/User.js]: Why is this if statement here

File: server/routes/users.js [hunk 2]
```diff
@@ -2,67 +2,101 @@ const { v4: uuidv4 } = require('uuid');
...
 })
 
 router.post('/teacher/', function(req, res, next) {
-    const emailQuery = User.where({email: req.body.email});
-    var user = new User({
-        name: req.body.displayName,
-        email: req.body.email,
-        password: req.body.password,
-        displayName: req.body.displayName,
+    var user = {
```
Comment (ZOrangeBandit) [inline server/routes/users.js]: A request's json is stored at req.body, not req. i.e. (req.email -> req.body.email)

File: server/routes/users.js [hunk 3]
```diff
@@ -2,67 +2,101 @@ const { v4: uuidv4 } = require('uuid');
...
         isTeacher: true,
+        sentMessageIDs: [],
+        recievedMessageIDs: [],
+        groupIDs: [],
+        active: false,
+        auth: {
+            token: ''
+        }
+    }
+    db.createTeacher(user)
```
Comment (ZOrangeBandit) [inline server/routes/users.js]: This usage of the createTeacher method is not the useage defined in the DatabaseAccessor class

File: server/database/User.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                        }
+                        else {
+                            
+                        }
+                    })
+                }
+            })
+    }
+
+    static
```
Comment (ZOrangeBandit) [inline server/database/User.js]: This is unneccissary and should be removed

File: server/database/User.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        }
+        let user = {...defaults, ...userIn};
+        if (!user.name){
+            user.name = user.displayName;
+        }
+        if (!user.displayName){
+            user.displayName = user.name;
+        }
+
+        // if
```
Comment (ZOrangeBandit) [inline server/database/User.js]: remove this comment

**Review by ZOrangeBandit — CHANGES_REQUESTED** [review]
There are a couple issues and peculiar changes that I think should be addressed before merging

**Review by ZOrangeBandit — APPROVED** [review]
Seems fine to me

File: server/database/User.js (hunk 1 above)
Comment (turtle11235) [reply server/database/User.js]: I actually meant to delete that whole folder

**Pull request merged**